src/main/java/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/machines/electric/AutomatedCraftingChamber.java

```diff
@@ -192,40 +198,65 @@ public boolean isSynchronized() {
 			}
 		});
 	}
-	
+
 	protected void tick(Block b) {
 		if (BlockStorage.getLocationInfo(b.getLocation(), "enabled").equals("false")) return;
 		if (ChargableBlock.getCharge(b) < getEnergyConsumption()) return;
-		
+
 		BlockMenu menu = BlockStorage.getInventory(b);
-		
+
+		String input = getSerializedInput(menu, false);
+		testInputAgainstRecipes(b, input);
+	}
+
+	protected void tickOnce(Block b) {
+		if (ChargableBlock.getCharge(b) < getEnergyConsumption()) return;
+
+		BlockMenu menu = BlockStorage.getInventory(b);
+
+		String input = getSerializedInput(menu, true);
+		testInputAgainstRecipes(b, input);
+	}
+
+	private String getSerializedInput(BlockMenu menu, boolean craftLast) {
 		StringBuilder builder = new StringBuilder();
 		int i = 0;
 		for (int j = 0; j < 9; j++) {
 			if (i > 0) {
 				builder.append(" </slot> ");
 			}
-			
+
 			ItemStack item = menu.getItemInSlot(getInputSlots()[j]);
-			if (item != null && item.getAmount() == 1) return;
+			if (craftLast) {
+				if (item != null && item.getAmount() > 1) return "";
+			} else {
+				// we're only executing the last possible shaped recipe
+				// we don't want to allow this to be pressed instead of the default timer-based
+				//   execution to prevent abuse and auto clickers
+				if (item != null && item.getAmount() == 1) return "";
+			}
+
 			builder.append(CustomItemSerializer.serialize(item, ItemFlag.MATERIAL, ItemFlag.ITEMMETA_DISPLAY_NAME, ItemFlag.ITEMMETA_LORE));
-			
+
 			i++;
 		}
-		
-		String input = builder.toString();
-		
+
+		return builder.toString();
+	}
+
+	private void testInputAgainstRecipes(Block block, String input) {
+		BlockMenu menu = BlockStorage.getInventory(block);
+
 		if (SlimefunPlugin.getUtilities().automatedCraftingChamberRecipes.containsKey(input)) {
 			ItemStack output = SlimefunPlugin.getUtilities().automatedCraftingChamberRecipes.get(input).clone();
-			
+
 			if (menu.fits(output, getOutputSlots())) {
 				menu.pushItem(output, getOutputSlots());
-				ChargableBlock.addCharge(b, -getEnergyConsumption());
```

Comment:
@TheBusyBiscuit power consumption is handled here, the method is called for both tick() and tickOne()

Reply:
Yeah, you can tell that I just woke up...